ResearchKit/+Medable/ORKMDBarcodeScannerStep/ORKMDBarcodeScannerStepViewController.m

```diff
@@ -0,0 +1,291 @@
+
+- (instancetype)initWithStep:(ORKStep *)step
+{
+    self = [super initWithStep:step];
+    
+    if (self)
+    {
+        NSParameterAssert([step isKindOfClass:[ORKMDBarcodeScannerStep class]]);
+    }
+    
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    [self stepDidChange];
+    
+    self.instructionStepView = [[ORKInstructionStepView alloc] initWithFrame:self.view.bounds];
+    self.instructionStepView.autoresizingMask = (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
+    [self.view addSubview:self.instructionStepView];
+    
+    if (ORKMDBarcodeScannerView.isSuppported)
+    {
+        CGRect scannerFrame = self.view.bounds;
+        scannerFrame.size.height *= 0.6;
+        
+        self.scannerView = [[ORKMDBarcodeScannerView alloc] initWithFrame:scannerFrame];
+        self.scannerView.autoresizingMask = (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
+        self.scannerView.delegate = self;
+        [self.view addSubview:self.scannerView];
+
+        // if we don't set these text labels of ORKInstructionStepView
+        // to _something_ here in viewDidLoad, it isn't smart enough
+        // to do the right thing when these values are set later.
+        self.instructionStepView.headerView.captionLabel.text = @" ";
+        self.instructionStepView.headerView.instructionLabel.text = @" ";
+    }
+    else
+    {
+        [self enableSkip];
+        self.configurationDone = YES;
+        
+        self.instructionStepView.headerView.captionLabel.text = @"Scanning Unavailble";
+        
+        [self.instructionStepView.headerView.instructionLabel setText:
+         @"Barcode scanning is supported on iOS 10 and above.  Please update your iOS version on your device."];
+    }
+    
+    UIImage* image = [UIImage imageNamed:@"barcode overlay"
+                                inBundle:ORKBundle()
+                                compatibleWithTraitCollection:nil];
+    
+    UIImageView *overlayImage = [[UIImageView alloc] initWithImage:
+                                 [image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]];
+    
+    [self.scannerView addSubview:overlayImage];
+    overlayImage.frame = self.scannerView.bounds;
+    overlayImage.backgroundColor = UIColor.clearColor;
+    overlayImage.tintColor = [UIColor colorWithRed:1.0
+                                             green:204.0/255.0
+                                              blue:0 alpha:1.0];
+    overlayImage.contentMode = UIViewContentModeScaleAspectFill;
+    
+    self.scannerView.accessibilityHint = self.barcodeScannerStep.accessibilityInstructions;
+}
+
+- (void)viewDidAppear:(BOOL)animated
+{
+    [super viewDidAppear:animated];
+    
+    if (self.configurationDone)
+    {
+        if (self.scannerView.isConfigured)
+        {
+            [self.scannerView startScanning];
+        }
+        else [self enableSkip]; // back navigation
```

Comment:
Recommend using braces even though this is a single line statement, just to make it clear what's going o.